pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,143 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
+        boolean booleanValue = getBoolean(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(booleanValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Short.class) {
+      if (sqlType == Types.SMALLINT) {
+        short shortValue = getShort(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(shortValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Integer.class) {
+      if (sqlType == Types.SMALLINT || sqlType == Types.INTEGER) {
+        int intValue = getInt(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(intValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Long.class) {
+      if (sqlType == Types.BIGINT) {
+        long longValue = getLong(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(longValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Float.class) {
+      if (sqlType == Types.REAL) {
+        float floatValue = getFloat(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(floatValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Double.class) {
+      if (sqlType == Types.FLOAT || sqlType == Types.DOUBLE) {
+        double doubleValue = getDouble(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(doubleValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Date.class) {
+      if (sqlType == Types.DATE) {
+        return type.cast(getDate(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Time.class) {
+      if (sqlType == Types.TIME) {
+        return type.cast(getTime(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Timestamp.class) {
```

Comment:
B.4 JDBC specification alse mentions that java.util.Calendar and java.util.Date maps to TIMESTAMP.

Reply:
Done